src/ui/public/kbn_top_nav/kbn_top_nav_controller.js

```diff
@@ -53,13 +53,18 @@ export default function ($compile) {
 
     // apply the defaults to individual options
     _applyOptDefault(opt = {}) {
-      return defaults({}, opt, {
+      const defaultedOpt = defaults({}, opt, {
         label: capitalize(opt.key),
         hasFunction: !!opt.run,
         description: opt.run ? opt.key : `Toggle ${opt.key} view`,
-        hideButton: !!opt.hideButton,
-        run: (item) => this.toggle(item.key)
+        run: (item) => !item.disableButton() && this.toggle(item.key)
       });
+
+      defaultedOpt.hideButton = isFunction(opt.hideButton) ? opt.hideButton : () => (opt.hideButton || false);
```

Comment:
You can use [_.result](https://lodash.com/docs#result) for all of these.

For example: `defaultedOpt.hideButton = result(opt, 'hideButton', false);`